satellite/overlay/keylock_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package overlay_test
+
+import (
+	"strconv"
+	"testing"
+
+	"storj.io/storj/internal/teststorj"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/overlay"
+)
+
+func TestKeyLock(t *testing.T) {
+	ml := overlay.NewKeyLock()
+	key := teststorj.NodeIDFromString("hi")
+	unlockFunc := ml.Lock(key)
+	unlockFunc()
+	unlockFunc = ml.RLock(key)
+	unlockFunc()
+}
+
+func BenchmarkKeyLock(b *testing.B) {
+	b.ReportAllocs()
+	ml := overlay.NewKeyLock()
+	numNodes := 100000
+	nodeIDs := make([]storj.NodeID, numNodes)
+	for i := 0; i < numNodes; i++ {
+		nodeIDs[i] = teststorj.NodeIDFromString(strconv.Itoa(i))
+	}
+	b.Run("lock", func(b *testing.B) {
```

Comment:
Separate the measurement of the first lock and relocking.

Reply:
I'm not sure what you mean here, can you explain further?

Reply:
I suspect there's a performance difference for adding a new node and locking/unlocking an already existing node.

Most of the time we will be locking/unlocking existing node-ids rather than adding, so we care about the latter number more. It's nicer if these benchmarks show up separately.

Reply:
I made that change. Just FYI, here's the results:

```
BenchmarkKeyLock/lock_all_new_nodes-8            2000000               805 ns/op
BenchmarkKeyLock/lock_existing_nodes-8          10000000               177 ns/op
BenchmarkKeyLock/rlock_existing_nodes-8         10000000               166 ns/op
```

Reply:
nice :)